pyartifactory/objects.py

```diff
@@ -399,31 +407,99 @@ def delete(self, name: str) -> None:
         """
         self.get(name)
         self._delete(f"api/{self._uri}/{name}")
-        logging.debug("Group %s successfully deleted", name)
+        logger.debug("Group %s successfully deleted", name)
 
 
 class ArtifactoryRepository(ArtifactoryObject):
     """Models an artifactory repository."""
 
     _uri = "repositories"
 
+    # Repositories operations
+    def get_repo(self, repo_name: str) -> AnyRepositoryResponse:
+        """
+        Finds repository in artifactory. Raises an exception if the repo doesn't exist.
+        :param repo_name: Name of the repository to retrieve
+        :return: Either a local, virtual or remote repository
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{repo_name}")
+            repo: AnyRepositoryResponse = parse_obj_as(
+                Union[
+                    LocalRepositoryResponse,
+                    VirtualRepositoryResponse,
+                    RemoteRepositoryResponse,
+                ],
+                response.json(),
+            )
+            return repo
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404 or error.response.status_code == 400:
+                logger.error("Repository %s does not exist", repo_name)
+                raise RepositoryNotFoundException(
+                    f" Repository {repo_name} does not exist"
+                )
+            raise ArtifactoryException from error
+
+    def create_repo(
+        self, repo: AnyRepositoryResponse
+    ) -> Union[
+        LocalRepositoryResponse, VirtualRepositoryResponse, RemoteRepositoryResponse
+    ]:
```

Comment:
Errrm, I think it's supposed to be the other way around. The previous code was
```python
    def create_repo(
        self, repo: Union[LocalRepository, VirtualRepository, RemoteRepository]
    ) -> Union[
        LocalRepositoryResponse, VirtualRepositoryResponse, RemoteRepositoryResponse
    ]:
```
so you would change it to
```suggestion
    def create_repo(
        self, repo: Union[LocalRepository, VirtualRepository, RemoteRepository]
    ) -> AnyRepositoryResponse:
```
(You could also define an `AnyRepository` alias, in which case I would probably move them in the `models` package)